wilson/util/common.py

```diff
@@ -0,0 +1,937 @@
+    _symmetry_class_defintions = {
+        # The keys are sorted n-tuples of 2-tuples in which the first entry is
+        # a string specifying the indices of a Wilson coefficient in the non-
+        # redundant two-flavour basis and the second entry is True (False) if
+        # the Wilson coefficient is real (complex). The values are integer ids
+        # denoting the symmetry class.
+        (): 0, # 0F scalar object
+        (
+            ('11', False), ('12', False), ('21', False), ('22', False),
+        ): 1, # 2F general matrix
+        (
+            ('11', True), ('12', False), ('22', True),
+        ): 2, # 2F Hermitian matrix
+        (
+            ('11', False), ('12', False), ('22', False),
+        ): 9, # 2F symmetric matrix
+        (
+            ('1111', True), ('1112', False), ('1122', True), ('1212', False),
+            ('1221', True), ('1222', False), ('2222', True),
+        ): 4, # 4F two identical ffbar currents, hermitian
+        (
+            ('1111', False), ('1112', False), ('1121', False), ('1122', False),
+            ('1212', False), ('1221', False), ('1222', False), ('2121', False),
+            ('2122', False), ('2222', False),
+        ): 41, # 4F two identical ffbar currents, non-hermitian
+        (
+            ('1111', True), ('1112', False), ('1122', True), ('1211', False),
+            ('1212', False), ('1221', False), ('1222', False), ('2211', True),
+            ('2212', False), ('2222', True),
+        ): 5, # 4F two independent ffbar currents
+        (
+            ('1111', True), ('1112', False), ('1122', True), ('1212', False),
+            ('1222', False), ('2222', True),
+        ): 6, # 4F two identical ffbar currents - special case Cee
+        (
+            ('1111', False), ('1112', False), ('1121', False), ('1122', False),
+            ('1211', False), ('1212', False), ('1221', False), ('1222', False),
+            ('2111', False), ('2112', False), ('2121', False), ('2122', False),
+            ('2211', False), ('2212', False), ('2221', False), ('2222', False),
+        ): 3, # 4F general four-index object
+        (
+            ('1111', False), ('1112', False), ('1121', False), ('1122', False),
+            ('1211', False), ('1212', False), ('1221', False), ('1222', False),
+            ('2211', False), ('2212', False), ('2221', False), ('2222', False),
+        ): 7, # 4F symmetric in first two indices
+        (
+            ('1211', False), ('1212', False), ('1221', False), ('1222', False),
+        ): 71, # 4F antisymmetric in first two indices
+        (
+            ('1111', False), ('1112', False), ('1121', False), ('1122', False),
+            ('1211', False), ('1212', False), ('1221', False), ('1222', False),
+            ('2121', False), ('2122', False), ('2221', False), ('2222', False),
+        ): 8, # 4F Baryon-number-violating - special case Cqqql
+    }
+
+    def __init__(self, eft, basis, dim4_keys_shape, dim4_symm_keys, n_gen = 3):
+        self._eft = eft
```

Comment:
Just curious, why are those underscored? `this.arg = arg` is a very common pattern...

Reply:
I think I get it, you are trying to precisely reproduce the interface of the old module, clever! :)

Reply:
Yes, I wanted to reproduce the old interface and make only those attributes public that have been used so far. But in the end, I think it doesn't really matter much and also the arguments of `__init__` could be made public by removing the underscores if they are a bit confusing at the moment.

Reply:
I will still keep `self._dim4_keys_shape` and `self._dim4_symm_keys` since they might be removed at a later point if we could add this information on the dim4 parameters to the WCxf basis definition.